### PR TITLE
Mlflow setup

### PR DESCRIPTION
**Pull request by RobotPsychologist**
Reconfigured the training run set up to create organized experiment and training runs for the ttm fine tuning. It now has a model registry (some bug fixes needed regarding trackin gthe loss metrics in the csv), some light documentation explaining how it works (WIP).

Removed some packaged from the requirments.txt, mostly the ones related to tensorflow because they were causing serious problems and slowdowns with GPU utilization, interfering with pytorch. 

A whole bunch of model results were merged for now so that others can see how the results turn out and the csv DB works.

### CODE REVIEW

File: scripts/watgpu/ttm_finetune.sh [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+ELAPSED_HOURS=$((ELAPSED_TIME / 3600))
+ELAPSED_MINUTES=$(((ELAPSED_TIME % 3600) / 60))
+ELAPSED_SECONDS=$((ELAPSED_TIME % 60))
+
+echo "Training finished at: $RUN_END_TIMESTAMP"
+echo "Training exit code: $TRAIN_EXIT_CODE"
+echo "Total elapsed time: ${ELAPSED_HOURS}h ${ELAPSED_MINUTES}m ${ELAPSED_SECONDS}s ($ELAPSED_TIME seconds)"
+
+# Stop GPU monitoring
+kill $GPU_MONITOR_PID $GPU_UTIL_PID 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply scripts/watgpu/ttm_finetune.sh:206]: Good catch, the required change was hire up, GPU_UTIL_PID was being defined as GPU_MON_PID instead.

File: src/train/ttm.py [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+    )
+    # datetime has to be a column
+    data = all_patients_df.reset_index()
+
+    # Train/val split
+    train, test = data_split
+    split_config = {"train": train, "test": test}
+    data_length = len(data)
+    info_print(f"Data length: {data_length}")
+    info_print(f"Split config: Train {train * 100}%, Test {test * 100}%")
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply src/train/ttm.py:546]: I changed ttm_custom_metrics.py instead.

**Review by Tony911029 — APPROVED** [review]
LGTM. Probably should name out .out better

File: cache/data/kaggle_brisT1D/processed/splits/split_bfff951e.pkl
Comment (Tony911029) [inline cache/data/kaggle_brisT1D/processed/splits/split_bfff951e.pkl:1]: Should this even be here?

File: scripts/watgpu/ttm_finetune.sh [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+# Create run summary for registry
+echo "=== Run Summary ===" >> "$RUN_DIR/run_info.txt"
+echo "Exit Code: $TRAIN_EXIT_CODE" >> "$RUN_DIR/run_info.txt"
+echo "Status: $([ $TRAIN_EXIT_CODE -eq 0 ] && echo 'completed' || echo 'failed')" >> "$RUN_DIR/run_info.txt"
+echo "=====================================" >> "$RUN_DIR/run_info.txt"
+
+# Update model registry with completion info
+echo "Updating model registry with completion info..."
+python -c "
```
Comment (Tony911029) [inline scripts/watgpu/ttm_finetune.sh:231]: is there any reason we are running an inline Python script? Doesn't seem to safe too me

File: results/runs/model_registry.py [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+import csv
+import os
+from datetime import datetime
+from pathlib import Path
+
+import pandas as pd
+
+
+class ModelRegistry:
+    def __init__(self, registry_path="results/runs/ttm_finetune/model_registry.csv"):
```
Comment (Tony911029) [inline results/runs/model_registry.py:16]: I thought ML flow is supposed to do all these? We can probably stick some metadata in, like server resource config to it no?
